Lib/functools.py

```diff
@@ -226,6 +222,45 @@ def __ge__(self, other):
     pass
 
 
+################################################################################
+### reduce() sequence to a single item
+################################################################################
+
+_initial_missing = object()
```

Comment:
functools.py already has a sentinel singleton, _NOT_FOUND used by cached_property. Would it be possible to reuse it, instead of adding a second sentinel value?

Reply:
Not sure how I feel about this suggestion. On the one hand, there would be a (slight) overhead reduction by reusing the same sentinel. On the other, semantically, this isn't really a `_NOT_FOUND`, so much as a "not given".

As these sentinels aren't part of the public interface, we could rename them into an object that semantically makes sense - but I'm loathe to touch another function, if I can help it.

Reply:
Ok, keep _initial_missing.